src/syft/grid/network.py

```diff
@@ -2,25 +2,54 @@
 functionality on an actual local network. This is NOT meant to be run in
 production (that's the *actual* grid's job)."""
 
-# stdlib
-import pickle
 
 # third party
 from flask import Flask
+from flask import request
+from nacl.encoding import HexEncoder
 
 # syft relative
+from ..core.common.message import SignedImmediateSyftMessageWithReply
+
+# relative absolute
+from ..core.common.message import SignedImmediateSyftMessageWithoutReply
+from ..core.common.serde.deserialize import _deserialize
 from ..core.node.network.network import Network
+from ..grid.services.signaling_service import PullSignalingService
+from ..grid.services.signaling_service import PushSignalingService
 
 app = Flask(__name__)
 
-network = Network(name="ucsf-net")
+network = Network(name="om-net")
+
+network.immediate_services_without_reply.append(PushSignalingService)
+network.immediate_services_with_reply.append(PullSignalingService)
+network._register_services()  # re-register all services including SignalingService
+
+
+@app.route("/metadata")
+def get_metadata() -> str:
+    return network.get_metadata_for_client()
 
 
-@app.route("/")
-def get_client() -> str:
-    client = network.get_client()
-    return pickle.dumps(client).hex()
+@app.route("/", methods=["POST"])
+def process_network_msgs() -> str:
+    json_msg = request.get_json()
+    obj_msg = _deserialize(blob=json_msg, from_json=True)
+    if isinstance(obj_msg, SignedImmediateSyftMessageWithReply):
+        reply = network.recv_immediate_msg_with_reply(msg=obj_msg)
+        return reply.json()
+    elif isinstance(obj_msg, SignedImmediateSyftMessageWithoutReply):
+        network.recv_immediate_msg_without_reply(msg=obj_msg)
+    else:
+        network.recv_eventual_msg_without_reply(msg=obj_msg)
+    return ""
 
 
 def run() -> None:
+    global network
+    print("====================================")
+    print("========== NODE ROOT KEY ===========")
+    print("====================================")
+    print(network.signing_key.encode(encoder=HexEncoder).decode("utf-8"), "\n")
```

Comment:
If I'm not mistaken, this source file should be just a "dev" version of the real network (which will be implemented in PyGrid repository in order to provide ).
The idea is to provide an easy way to test/use an instance (without worrying about scalability, fault tolerance, etc.)
So, I think there's no problem to show the root key in this case (since this file will not be used as a real server).

But it may be cool to add a disclaimer stating this. :)

Reply:
Totally - this is just to empower light and local dev over an actual (local) network

Reply:
I have added a comment explaining that printing of the signing_key is expected and not related to the real PyGrid implementation.